src/org/zaproxy/zap/extension/openapi/generators/BodyGenerator.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.openapi.generators;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.swagger.models.properties.Property;
+import io.swagger.models.properties.RefProperty;
+
+public class BodyGenerator {
+
+    private ModelGenerator modelGenerator;
+    private DataGenerator dataGenerator;
+
+    public BodyGenerator(ModelGenerator modelGenerator, DataGenerator dataGenerator) {
+        this.modelGenerator = modelGenerator;
+        this.dataGenerator = dataGenerator;
+    }
+
+    private enum Element {
+        object_begin,
+        object_end,
+        array_begin,
+        array_end,
+        property_container,
+        inner_separator,
+        outer_separator
+    }
+
+    @SuppressWarnings("serial")
+    private final Map<Element, String> SYNTAX = Collections.unmodifiableMap(new HashMap<Element, String>() {
+
+        {
+            put(Element.object_begin, "{");
+            put(Element.object_end, "}");
+            put(Element.array_begin, "[");
+            put(Element.array_end, "]");
+            put(Element.property_container, "\"");
+            put(Element.inner_separator, ":");
+            put(Element.outer_separator, ",");
+        }
+    });
+
+    public String generate(String name, boolean isArray) {

```

Comment:
This could be replaced with a JSON lib (instead of generating the JSON manually)? (guess it could be done later, it seems to be producing JSON correctly now).